oauthlib/common.py

```diff
@@ -197,7 +197,7 @@ def generate_token(length=30, chars=UNICODE_ASCII_CHARACTER_SET):
     return ''.join(rand.choice(chars) for x in range(length))
 
 
-def generate_signed_token(private_pem, request):
+def generate_signed_token(private_pem, request, algorithm='RS256', headers=None):
```

Comment:
we will need test coverage for this change

Reply:
@auvipy I don't see any tests for `generate_signed_token` so I can't test "this change". 

There's `signed_token_generator` usage in `oauthlib/tests/oauth2/rfc6749/test_server.py`. It would be easiest to extend `signed_token_generator` with new parameters and then use `SignedTokenEndpointTest` to add some tests for those new parameters. But technically speaking this would be testing `signed_token_generator` not `generate_signed_token`.

I'm open to suggestion on how to approach this. Also, how do I easily mock `request` parameter of `generate_signed_token` without getting into too many oauth details? This is a method from `common` so I shouldn't have to use any code that's nested under `rfc...` folders.

Reply:
foode222

Reply:
> @auvipy I don't see any tests for `generate_signed_token` so I can't test "this change". 
> 
> 
> 
> There's `signed_token_generator` usage in `oauthlib/tests/oauth2/rfc6749/test_server.py`. It would be easiest to extend `signed_token_generator` with new parameters and then use `SignedTokenEndpointTest` to add some tests for those new parameters. But technically speaking this would be testing `signed_token_generator` not `generate_signed_token`.
> 
> 
> 
> I'm open to suggestion on how to approach this. Also, how do I easily mock `request` parameter of `generate_signed_token` without getting into too many oauth details? This is a method from `common` so I shouldn't have to use any code that's nested under `rfc...` folders.

Reply:
Anyone? @auvipy?

Reply:
@JonathanHuot